streamlit/README.md

```diff
@@ -1,24 +1,28 @@
-## Streamlit Usage
+# Streamlit Usage
+## Demo App
 
-Start Streamlit local pages by navigating to the `data_parsing` directory and running the main script:
+[![Streamlit App](https://static.streamlit.io/badges/streamlit_badge_black_white.svg)](https://gsoc24-zebinhuang.streamlit.app/)
 
-```bash
-streamlit run streamlit/main.py
-```
+This repository is deployed on a personal Streamlit community cloud account.
 
 Note that Streamlit Community Cloud runs with the working directory set to /mount/src/gsoc-streamlit by default.
 
-This repository is deployed on a personal Streamlit community cloud account. You can directly access the page here: https://gsoc24-zebinhuang.streamlit.app/
+### Get an OpenAI API key
 
-When deploying the app, remember to include the OpenAI API key:
+You can get your own OpenAI API key by following the following instructions:
 
-```bash
-openai_api_key = "your_api_key"
-```
+1. Go to https://platform.openai.com/account/api-keys.
+2. Click on the `+ Create new secret key` button.
+3. Next, enter an identifier name (optional) and click on the `Create secret key` button.
 
-For more details on setting this up, refer to [Streamlit's discussion on setting OpenAI API keys](https://discuss.streamlit.io/t/struggling-with-setting-openai-api-using-streamlit-secrets/37959).
+## Run it locally
 
+```sh
+virtualenv .venv
+source .venv/bin/activate
+pip install -r requirements.txt
+streamlit run streamlit/main.py
+```
 
-Deployment Notes
 - The model training process may be slow due to the limitations of Streamlit Community Cloud.
-- The model used for testing is large and cannot be hosted directly on the GitHub repository. Currently, this model is not included in the Streamlit deployment.
+- The model ([See huggingface models](https://huggingface.co/zebin-huang/gsoc2024-ZebinHuang/tree/main)) used for testing is large and be hosted directly on the GitHub repository.
```

Comment:
This sentence is unclear. We could rewrite it 😄 

`The model used for testing is large and is hosted directly on the GitHub repository.`

Reply:
Hi @sergiopaniego, I've updated the documentation. Please let me know if you encounter any other issues. 😄